### PR TITLE
MOB-557 Isolated Positions: Add trade receipt items

### PR DESCRIPTION
**Pull request by ruixhuang**
Screenshot_1718906928

### CODE REVIEW

File: v4/feature/receipt/src/main/java/exchange/dydx/trading/feature/receipt/components/exchangerate/DydxReceiptExchangeRateViewModel.kt [hunk 1]
```diff
@@ -42,9 +42,9 @@ class DydxReceiptExchangeRateViewModel @Inject constructor(
         transferInput: TransferInput?
     ): String? {
         val exchangeRate = transferInput?.summary?.exchangeRate ?: return null
-        val type = transferInput?.type ?: return null
-        val token = transferInput?.token ?: return null
-        val symbol = transferInput?.resources?.tokenResources?.toMap()?.get(token)?.symbol ?: return null
+        val type = transferInput.type ?: return null
+        val token = transferInput.token ?: return null
+        val symbol = transferInput.resources?.tokenResources?.toMap()?.get(token)?.symbol ?: return null
```
Comment (prashanDYDX) [inline v4/feature/receipt/src/main/java/exchange/dydx/trading/feature/receipt/components/exchangerate/DydxReceiptExchangeRateViewModel.kt]: nit: `.toMap()` is unnecessary, IMap should have all the same methods

File: v4/feature/receipt/src/main/java/exchange/dydx/trading/feature/receipt/components/freecollateral/DydxReceiptFreeCollateralViewModel.kt [hunk 2]
```diff
@@ -2,113 +2,68 @@ package exchange.dydx.trading.feature.receipt.components.buyingpower
...
-                    }
-                    else -> flowOf()
+        abacusStateManager.marketId
+            .filterNotNull()
+            .flatMapLatest { marketId ->
+                combine(
+                    abacusStateManager.state.selectedSubaccountPositionOfMarket(marketId),
+                    abacusStateManager.state.selectedSubaccountPendingPositions.map { it?.firstOrNull { it.marketId == marketId } },
+                ) { position, pendingPosition ->
+                    Pair(position, pendingPosition)
```
Comment (prashanDYDX) [inline v4/feature/receipt/src/main/java/exchange/dydx/trading/feature/receipt/components/freecollateral/DydxReceiptFreeCollateralViewModel.kt]: nit: think we can just call `createViewState()` directly here.

also kotlin tip: `to` is a built in `infix fun` that creates Pairs. ex: `position to pendingPosition`

File: v4/feature/receipt/src/main/java/exchange/dydx/trading/feature/receipt/components/isolatedmargin/DydxReceiptIsolatedPositionMarginUsageViewModel.kt [hunk 3]
```diff
@@ -21,25 +28,83 @@ class DydxReceiptIsolatedPositionMarginUsageViewModel @Inject constructor(
...
+        return when (marginMode) {
+            MarginMode.Cross -> return notionalTotal?.postOrder?.let { notionalTotal ->
+                maintenanceMarginFraction?.let { maintenanceMarginFraction ->
+                    maintenanceMarginFraction * notionalTotal
+                }
+            }
+            MarginMode.Isolated -> equity?.postOrder
+            else -> null
+        }
+    }
```
Comment (prashanDYDX) [inline v4/feature/receipt/src/main/java/exchange/dydx/trading/feature/receipt/components/isolatedmargin/DydxReceiptIsolatedPositionMarginUsageViewModel.kt:106]: no change needed here, but these types of helper methods feel like they should be in abacus. (easy to make a mistake and needed on both platforms)

as we're building stuff in the future i think may be helpful to consider moving more things into abacus (not necessarily the state machine) to share between at least ios and android

File: v4/feature/shared/src/main/java/exchange/dydx/trading/feature/shared/viewstate/SharedMarketPositionViewState.kt [hunk 4]
```diff
@@ -159,9 +162,21 @@ data class SharedMarketPositionViewState(
                     position.liquidationPrice.current,
                     configs.displayTickSizeDecimals ?: 0,
                 ),
-                margin = position.valueTotal.current?.let {
-                    formatter.dollar(it.absoluteValue, 2)
-                } ?: formatter.dollar(position.notionalTotal.current?.absoluteValue, 2),
+                margin = when (position.marginMode) {
```
Comment (prashanDYDX) [inline v4/feature/shared/src/main/java/exchange/dydx/trading/feature/shared/viewstate/SharedMarketPositionViewState.kt:165]: we seem to be doing this calculation quite a lot. maybe extract to a helper?

**Review by prashanDYDX — APPROVED** [review]

File: v4/feature/receipt/src/main/java/exchange/dydx/trading/feature/receipt/components/isolatedmargin/DydxReceiptIsolatedPositionMarginUsageViewModel.kt (hunk 3 above)
Comment (ruixhuang) [reply v4/feature/receipt/src/main/java/exchange/dydx/trading/feature/receipt/components/isolatedmargin/DydxReceiptIsolatedPositionMarginUsageViewModel.kt:106]: Yes, already planned to move this to Abacus

File: v4/feature/receipt/src/main/java/exchange/dydx/trading/feature/receipt/components/exchangerate/DydxReceiptExchangeRateViewModel.kt (hunk 1 above)
Comment (ruixhuang) [reply v4/feature/receipt/src/main/java/exchange/dydx/trading/feature/receipt/components/exchangerate/DydxReceiptExchangeRateViewModel.kt]: Updated

File: v4/feature/receipt/src/main/java/exchange/dydx/trading/feature/receipt/components/freecollateral/DydxReceiptFreeCollateralViewModel.kt (hunk 2 above)
Comment (ruixhuang) [reply v4/feature/receipt/src/main/java/exchange/dydx/trading/feature/receipt/components/freecollateral/DydxReceiptFreeCollateralViewModel.kt]: Updated

**Pull request merged**